lib/TBDGen/TBDGen.cpp

```diff
@@ -482,7 +482,8 @@ void TBDGenVisitor::visitProtocolDecl(ProtocolDecl *PD) {
     auto isExpectedKind =
         isa<TypeAliasDecl>(member) || isa<AssociatedTypeDecl>(member) ||
         isa<AbstractStorageDecl>(member) || isa<PatternBindingDecl>(member) ||
-        isa<AbstractFunctionDecl>(member) || isa<IfConfigDecl>(member);
+        isa<AbstractFunctionDecl>(member) || isa<IfConfigDecl>(member) ||
+        isa<PoundDiagnosticDecl>(member);
     assert(isExpectedKind &&
            "unexpected member of protocol during TBD generation");
```

Comment:
Maybe we should just have a switch here. That way we won't forget anything in the future.